### PR TITLE
feat: probe for Encharge and Enpower support

### PR DESCRIPTION
**Pull request by cgarwood**
Probe for Encharge and Enpower devices and updated `supported_features`

### CODE REVIEW

File: src/pyenphase/envoy.py [hunk 1]
```diff
@@ -226,6 +228,27 @@ async def probe(self) -> None:
         else:
             self._supported_features |= SupportedFeatures.INVERTERS
 
+        # Check for various Ensemble support
+        if self._firmware.version >= ENSEMBLE_MIN_VERSION:
+            # The Ensemble Inventory endpoint will tell us if we have Enpower or Encharge support
+            try:
+                result = await self.request(URL_ENSEMBLE_INVENTORY)
+            except (json.JSONDecodeError, httpx.HTTPError) as e:
```
Comment (bdraco) [inline src/pyenphase/envoy.py:236]: We should probably make this a constant at some point

File: src/pyenphase/envoy.py [hunk 2]
```diff
@@ -226,6 +228,27 @@ async def probe(self) -> None:
...
+                _LOGGER.debug("Ensemble Inventory endpoint not found: %s", e)
+            else:
+                if result == []:
```
Comment (bdraco) [inline src/pyenphase/envoy.py:239]: maybe `if not result:`

**Review by bdraco — APPROVED** [review]

File: src/pyenphase/envoy.py (hunk 1 above)
Comment (cgarwood) [reply src/pyenphase/envoy.py:236]: What part? I think Github highlighted the wrong section

Comment (bdraco) [reply src/pyenphase/envoy.py:236]: (json.JSONDecodeError, httpx.HTTPError)

File: src/pyenphase/envoy.py [hunk 3]
```diff
@@ -230,6 +232,27 @@ async def probe(self) -> None:
         else:
             self._supported_features |= SupportedFeatures.INVERTERS
 
+        # Check for various Ensemble support
+        if self._firmware.version >= ENSEMBLE_MIN_VERSION:
+            # The Ensemble Inventory endpoint will tell us if we have Enpower or Encharge support
+            try:
+                result = await self.request(URL_ENSEMBLE_INVENTORY)
+            except (json.JSONDecodeError, httpx.HTTPError) as e:
```
Comment (bdraco) [inline src/pyenphase/envoy.py:240]:
```suggestion
            except ENDPOINT_PROBE_EXCEPTIONS as e:
```

**Pull request merged**